File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    
    def dfs(parent):
        visited[parent] = True
        for child in range(n):
            if computers[parent][child] == 1 and not visited[child]:
                dfs(child)
    
    for root in range(n):
        if not visited[root]:
            dfs(root)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const visited = new Array(n).fill(false);
    
    const dfs = (parent) => {
        visited[parent] = true;
        
        for (let child = 0; child < n; child++) {
            if (computers[parent][child] === 1 && visited[child] === false) {
                dfs(child);
            }
        }
    }
    
    for (let root = 0; root < n; root++) {
        if (visited[root] === false) {
            dfs(root);
            answer++;
        }
    }
    
    return answer;
}